x/auth/ante/sigverify.go

```diff
@@ -49,12 +50,14 @@ type SignatureVerificationGasConsumer = func(meter storetypes.GasMeter, sig sign
 // PubKeys must be set in context for all signers before any other sigverify decorators run
 // CONTRACT: Tx must implement SigVerifiableTx interface
 type SetPubKeyDecorator struct {
-	ak AccountKeeper
+	ak         AccountKeeper
+	signingCtx *txsigning.Context
 }
 
-func NewSetPubKeyDecorator(ak AccountKeeper) SetPubKeyDecorator {
+func NewSetPubKeyDecorator(ak AccountKeeper, signingCtx *txsigning.Context) SetPubKeyDecorator {
```

Comment:
should this ante handler get an AddressCodec instead of signingContext leaking all the way up here?

Reply:
We should probably have a fallback option in case it's nil rather than failing

Reply:
We can also use the address.Codec already on the AccountKeeper

Reply:
I added an `AddressCodec()` method to `AccountKeeper` and am using that instead. Let me know if that looks okay